### ISSUE TITLE
very slow when testing on megaface

### ISSUE THREAD

**Issue by twmht** [deepinsight/insightface#480]
Since the batch size is always 1 when testing megaface (https://github.com/deepinsight/insightface/blob/master/src/megaface/gen_megaface.py), is there any reason for setting such small batch size?

**Comment by nttstar** [comment]
I will provide a new megaface evaluation script.

**Comment by ruiming46zrm** [comment]
15hours spent running "gen_megaface.py"   ,  maybe multi thread needed

**Comment by nttstar** [comment]
Also the cropped facescrub images will be updated.

**Comment by twmht** [comment]
by the way, I use the provided LResNet34E-IR and only got 95.4235% ( but you got 96.7% with the same weight) on identification.

I use alignment code from insightface (https://github.com/deepinsight/insightface/tree/master/src/align) to align both facscrub and megaface.

**Comment by ruiming46zrm** [comment]
> by the way, I use the provided LResNet34E-IR and only got 95.4235% ( but you got 96.7% with the same weight) on identification.
> 
> I use alignment code from insightface (https://github.com/deepinsight/insightface/tree/master/src/align) to align both facscrub and megaface.

@twmht  Hi ,  what's your pretrained res50 rusult  by mega ?  mine is 0.965770

**Comment by ruiming46zrm** [comment]
@twmht  did you test the provided res50 by megaface? 
 it is said to be 97.64 ,  but my result is 0.965770

**Comment by nttstar** [comment]
Please use the new megaface testing tools at [https://github.com/deepinsight/insightface/tree/master/Evaluation/Megaface](https://github.com/deepinsight/insightface/tree/master/Evaluation/Megaface). Both facescrub and megaface images are aligned there.

**Comment by ruiming46zrm** [comment]
@nttstar  thank you

**Comment by ruiming46zrm** [comment]
@nttstar  Hi , today your provided aligned facescrub data is 92M , size same as original. 
but  you  once provided a pre-aligned facescrub data nearly 2G with more people: https://github.com/deepinsight/insightface/issues/167
can you tell  which to use  and why?

**Comment by twmht** [comment]
@nttstar 

looking forward for dropbox link.

**Comment by twmht** [comment]
@ruiming46zrm 

since the author said that they use matlab to align megaface, there are some differences between python's version. @nttstar  am I right?

**Comment by twmht** [comment]
@ruiming46zrm 

I tested resnet34 and also got 1% difference if I align the megaface with the provided python code.

**Comment by nttstar** [comment]
Some facescrub images must to be cropped manually.

**Comment by twmht** [comment]
@nttstar 

what do you mean cropped manually? Can you show the details in the document?

**Comment by nttstar** [comment]
Sometimes you will fail to crop the correct face from facescrub images with more than one person. You can also refer to here: [https://github.com/happynear/FaceVerification/tree/master/dataset/Megaface](https://github.com/happynear/FaceVerification/tree/master/dataset/Megaface). This is a must to get correct facescrub faces otherwise the testing accuracy will be random.

**Comment by twmht** [comment]
@nttstar 

Is the dropbox link available? baidu is too slow.

**Comment by ruiming46zrm** [comment]
@twmht   I tested  the downloaded new megaface align and facescrub align data , with provided model , 97.2% 
 but  pytroch version only get 87% with TreB1eN's model , and 85% with my self trained model,  can you share your gen_megaface.py in torch version?  and I may find where I am wrong

**Comment by twmht** [comment]
@ruiming46zrm 

the format of TreB1eN's model is RGB, and my test result is 90% for se-resnet50

**Comment by ruiming46zrm** [comment]
@twmht  you mean to delete the the  flipid added operation, right ?

**Comment by twmht** [comment]
@ruiming46zrm 

I keep the flipped operation. 

My mistake, TreB1eN's's model is BGR format. you should care about that.

**Comment by ruiming46zrm** [comment]
thanks~~

**Comment by nttstar** [comment]
dropbox link: https://www.dropbox.com/s/5ko2dcd1x7vn37w/megaface_testpack_v1.0.zip?dl=0
